app/slashcommands/command_invite.go

```diff
@@ -38,99 +37,166 @@ func (*InviteProvider) GetCommand(a *app.App, T i18n.TranslateFunc) *model.Comma
 				Text:         args.T("api.command_invite.channel.app_error"),
 				ResponseType: model.CommandResponseTypeEphemeral,
 			}
 		}
+		targetChannels = append(targetChannels, channelToJoin)
 	}
 
-	// Permissions Check
-	switch channelToJoin.Type {
-	case model.ChannelTypeOpen:
-		if !a.HasPermissionToChannel(c, args.UserId, channelToJoin.Id, model.PermissionManagePublicChannelMembers) {
-			return &model.CommandResponse{
-				Text: args.T("api.command_invite.permission.app_error", map[string]any{
-					"User":    userProfile.Username,
-					"Channel": channelToJoin.Name,
-				}),
-				ResponseType: model.CommandResponseTypeEphemeral,
-			}
-		}
-	case model.ChannelTypePrivate:
-		if !a.HasPermissionToChannel(c, args.UserId, channelToJoin.Id, model.PermissionManagePrivateChannelMembers) {
-			if _, err = a.GetChannelMember(c, channelToJoin.Id, args.UserId); err == nil {
-				// User doing the inviting is a member of the channel.
+	return targetUsers, targetChannels, nil
+}
+
+func (i *InviteProvider) getUserProfile(a *app.App, username string) *model.User {
+	userProfile, nErr := a.Srv().Store().User().GetByUsername(username)
+	if nErr != nil {
+		return nil
+	}
+
+	if userProfile.DeleteAt != 0 {
+		return nil
+	}
+
+	return userProfile
+}
+
+func (i *InviteProvider) checkPermissions(a *app.App, c request.CTX, args *model.CommandArgs, targetUser *model.User, targetChannels []*model.Channel) *model.CommandResponse {
+	var err *model.AppError
+	for _, targetChannel := range targetChannels {
+		switch targetChannel.Type {
+		case model.ChannelTypeOpen:
+			if !a.HasPermissionToChannel(c, args.UserId, targetChannel.Id, model.PermissionManagePublicChannelMembers) {
 				return &model.CommandResponse{
 					Text: args.T("api.command_invite.permission.app_error", map[string]any{
-						"User":    userProfile.Username,
-						"Channel": channelToJoin.Name,
+						"User":    targetUser.Username,
+						"Channel": targetChannel.Name,
 					}),
 					ResponseType: model.CommandResponseTypeEphemeral,
 				}
 			}
-			// User doing the inviting is *not* a member of the channel.
+		case model.ChannelTypePrivate:
+			if !a.HasPermissionToChannel(c, args.UserId, targetChannel.Id, model.PermissionManagePrivateChannelMembers) {
+				if _, err = a.GetChannelMember(c, targetChannel.Id, args.UserId); err == nil {
+					// User doing the inviting is a member of the channel.
+					return &model.CommandResponse{
+						Text: args.T("api.command_invite.permission.app_error", map[string]any{
+							"User":    targetUser.Username,
+							"Channel": targetChannel.Name,
+						}),
+						ResponseType: model.CommandResponseTypeEphemeral,
+					}
+				}
+				// User doing the inviting is *not* a member of the channel.
+				return &model.CommandResponse{
+					Text: args.T("api.command_invite.private_channel.app_error", map[string]any{
+						"Channel": targetChannel.Name,
+					}),
+					ResponseType: model.CommandResponseTypeEphemeral,
+				}
```

Comment:
When a user isn't authorized to manage private channel members, I can see we have a different error message for members vs non-members. I'm assuming that's a security measure to avoid revealing the existence of the channel? I think that because we're returning the error `api.command_invite.channel.error` earlier (in the `parseMessage` function) when a channel _actually_ can't be found, we're unintentionally proving that the channel exists by returning another message here.

Am I right or did I read or follow this logic incorrectly?